src/client/features/search/index.js

```diff
@@ -42,19 +45,51 @@ class Search extends React.Component {
   getSearchResult() {
     const state = this.state;
     const query = state.query;
-
     if (query.q !== '') {
       this.setState({
         loading: true
       });
+      this.getLandingResult();
       ServerAPI.querySearch(query)
         .then(searchResults => {
+           this.setState({
+             searchResults: searchResults,
+             loading: false
+           });
+        });
+    }
+  }
+
+  getLandingResult() {
+    const state = this.state;
+    const query = {
+      q: state.query.q,
+        gt: -1,
+        lt: 250,
+        type: 'ProteinReference',
+        datasource: state.query.datasource,
+        species: '9606'
+    };
+    this.setState({
```

Comment:
setState is asynchronous. to gaurantee that the code after this runs in sequence you should do something like:

```js
this.setState({
  landingLoading: true
}, () => {
  ServerAPI.findUniprotId(...)...
});

```

Reply:
you can optionally supply a callback function that runs after setState has completed